join.spec.md

```diff
@@ -131,48 +131,42 @@ Documents MUST define a {join__Graph} enum. Each enum value defines a subgraph.
 
 The {join__Graph} enum is used as input to {@join} directives which link fields and types to subgraphs.
 
-# Scalars
-
-Documents MUST include definitions of any scalars they use. Producers SHOULD include the `@specifiedBy` directive, but it is not mandatory.
-
-##! join__Never
+# Directives
 
-:::[definition](photos.graphql#join__Never)
+##! @join__graph
 
-The {join__Never} scalar is used to indicate the type of a field which **cannot be resolved**. This is typically due to a composition error. Producers SHOULD provide more specific error information with an {@error} directive on the relevant field.
+Declare subgraph metadata on {join__Graph} enum values. 
 
-:::[example](photos.graphql#User.favorite) -- Using {join__Never} and {@error} to mark a field error
+```graphql definition
+directive @join__graph(name: String!, url: String!) on ENUM_VALUE
+```
 
-# Directives
+:::[example](photos.graphql#join__Graph) -- Using {@join__graph} to declare subgraph metadata on the {join__Graph} enum values.
 
 ##! @join__type
 
 Join a type to a subgraph, optionally providing an entity key.
 
 ```graphql definition
 directive @join__type(
-  graph: join__Graph!,
-  key: ref__Fragment)
-  repeatable on OBJECT
+  graph: join__Graph!
+  key: String
+) repeatable on OBJECT | INTERFACE
 ```
 
+// TODO: rename portal query?
 Keys will be passed as `representations` within a [portal query](#portal-query) to [port](#portability) a selection set between subgraphs.
 
 :::[example](photos.graphql#Image) -- Using {@join__type} to specify subgraph keys
 
-```html diagram
-<script>line => elStr(line).includes("key:")</script>
-```
-
 Multiple {@join}s can be specified for different subgraphs. It is an error to {@join} an object against the same subgraph multiple times.
```

Comment:
Be more literal in the second sentence: "It is an error to place two {@join__type} directives on the same type with the same `graph` argument." or maybe use some sort of MUST NOT rather than "it is an error"

(Though is that true? I see on https://github.com/apollographql/federation/issues/624 that @martijnwalraven has crossed off my claim that we need to document this and said "I don't believe this is what we want".)

Reply:
Fixed on my branch